### ISSUE TITLE
Add raster sum and multiply functions

### ISSUE THREAD

**Issue by emlys** [natcap/pygeoprocessing#234]
We often sum and multiply rasters using `raster_calculator`. This is easy enough but the sum and multiply functions are re-implemented in at least a couple different models. Allison recently asked about examples of how to sum rasters using pygeoprocessing. It could be nice to give these basic operations their own pygeoprocessing functions. 
(also consider: count and mean operations)

**Comment by phargogh** [comment]
It would be nice to de-duplicate these in InVEST!  We should be sure to take a look at where the functions would be used and design the common solution accordingly.

**Comment by dcdenu4** [comment]
I agree that if there is something running up against our usual rule of 3 we should take a look at common solution. I'll note that in Allison's case she wanted a sum by pixel where `nodata` values were treated as 0.
